### PR TITLE
Add SMTP configuration for alerting.

### PR DESCRIPTION
**Pull request by mreid-moz**
Per bug 1274968, setup smtp for alert delivery.

---
This change is [Reviewable](https://reviewable.io/reviews/vitillo/telemetry-airflow/9)

### CODE REVIEW

File: ansible/files/airflow/entrypoint.sh [hunk 1]
```diff
@@ -28,6 +29,27 @@ sed -i "s/{DB_USER}/${DB_USER}/" $AIRFLOW_HOME/airflow.cfg
 sed -i "s/{DB_PASSWORD}/${DB_PASSWORD}/" $AIRFLOW_HOME/airflow.cfg
 sed -i "s/{AIRFLOW_ENABLE_AUTH}/${AIRFLOW_ENABLE_AUTH}/" $AIRFLOW_HOME/airflow.cfg
 
+# Fetch the SMTP credentials
```
Comment (vitillo) [inline ansible/files/airflow/entrypoint.sh]: Could we fetch the credentials from the ansible script and pass them as environment variables?

**Comment by mreid-moz** [comment]
*[ansible/files/airflow/entrypoint.sh, line 32 \[r1\]](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-KLMHOgZBy7a6gleU3nr-r1-32:-KLMQTXZ3Wx8RWOEYu3z:2090078797) ([raw file](https://github.com/vitillo/telemetry-airflow/blob/c3c1ea87cb016b533c04b4b4948568a0b2dbfb42/ansible/files/airflow/entrypoint.sh#L32)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> Could we fetch the credentials from the ansible script and pass them as environment variables?
</details>
This would be better. Looks like the `aws` command is not available inside the container.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9)*

**Comment by mreid-moz** [comment]
*[ansible/files/airflow/entrypoint.sh, line 32 \[r1\]](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-KLMHOgZBy7a6gleU3nr-r1-32:-KLMaCKUgVvfKEHBlfjN:1497076627) ([raw file](https://github.com/vitillo/telemetry-airflow/blob/c3c1ea87cb016b533c04b4b4948568a0b2dbfb42/ansible/files/airflow/entrypoint.sh#L32)):*
<details><summary><i>Previously, mreid-moz (Mark Reid) wrote…</i></summary>
> This would be better. Looks like the `aws` command is not available inside the container.
</details>
Ok, I've changed things around to fetch and populate the credentials using ansible, rather than within the container itself. To test the S3 fetch / JSON parse, I used the following playbook:

```yaml
- name: Testing testing
  hosts: localhost
  vars:
    elb_name: telemetry-ecs
    command: scheduler

  tasks:
    - name: fetch SMTP credentials
      s3: bucket={{ metadata_bucket }} region={{region}} object=smtp_ses/credentials.json dest=smtp_credentials.json mode=getstr
      register: smtp_credentials_raw
      
    - set_fact:
        smtp_credentials: "{{ smtp_credentials_raw.contents|from_json }}"

    - name: create task definition
      shell: "echo '{{ smtp_credentials.user }}'"
```

I also verified that the "dummy" values are passed through correctly from `deploy_local.yml`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9)*

**Comment by vitillo** [comment]
Has this patch been tested to see if Airflow sends e-mails on failure?

---

Review status: 0 of 7 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-:-KLMi44IZFniPM2HGZmw:-1541393806)*

**Comment by mreid-moz** [comment]
I've now tested locally and airflow can send email! Turns out one of the config parmeters [needed to be changed](https://github.com/apache/incubator-airflow/blob/master/UPDATING.md#email-configuration-change) in addition to setting up correct SMTP credentials.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-:-KLNJwC4FUoAk6RWDcLl:1968529762)*

**Comment by vitillo** [comment]
Reviewed 5 of 6 files at r2, 1 of 1 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[ansible/deploy_aws.yml, line 41 \[r4\]](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-KLPgQ0Mv58T4zuHx-AO:-KLPgQ0Mv58T4zuHx-AP:-1686147130) ([raw file](https://github.com/vitillo/telemetry-airflow/blob/38dc9df56d76111e8ff82fb242a23290da941283/ansible/deploy_aws.yml#L41)):*
> ```YAML
>     - name: fetch SMTP credentials
>       s3: bucket={{metadata_bucket}} region={{region}} object=smtp_ses/credentials.json dest=smtp_credentials.json mode=getstr
>       register: smtp_credentials_json
> ```

Is `dest` required when using `mode=getstr`?

---

*[ansible/envs/dev.yml, line 28 \[r4\]](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-KLPgvpHZnnycfk4pTsT:-KLPgvpHZnnycfk4pTsU:320997331) ([raw file](https://github.com/vitillo/telemetry-airflow/blob/38dc9df56d76111e8ff82fb242a23290da941283/ansible/envs/dev.yml#L28)):*
> ```YAML
> 
> airflow_enable_auth: false
> metadata_bucket: net-mozaws-prod-us-west-2-pipeline-metadata
> ```

Is this variable ever accessed in the dev environment?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9)*

**Comment by mreid-moz** [comment]
*[ansible/envs/dev.yml, line 28 \[r4\]](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-KLPgvpHZnnycfk4pTsT:-KLRNewVNa1ZtK2-KB9S:-2144293953) ([raw file](https://github.com/vitillo/telemetry-airflow/blob/38dc9df56d76111e8ff82fb242a23290da941283/ansible/envs/dev.yml#L28)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> Is this variable ever accessed in the dev environment?
</details>
No, I'll remove it

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9)*

**Comment by mreid-moz** [comment]
*[ansible/deploy_aws.yml, line 41 \[r4\]](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-KLPgQ0Mv58T4zuHx-AO:-KLRNqNXLbxCFZKuoX4J:-1448331212) ([raw file](https://github.com/vitillo/telemetry-airflow/blob/38dc9df56d76111e8ff82fb242a23290da941283/ansible/deploy_aws.yml#L41)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> Is `dest` required when using `mode=getstr`?
</details>
Nope! I left that in from the first pass where I actually fetched the file. I'll remove it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9)*

**Comment by mreid-moz** [comment]
Review status: 3 of 8 files reviewed at latest revision, 2 unresolved discussions.

---

*[ansible/deploy_aws.yml, line 41 \[r4\]](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-KLPgQ0Mv58T4zuHx-AO:-KLRQte4G3Fe27chqxvx:-499146009) ([raw file](https://github.com/vitillo/telemetry-airflow/blob/38dc9df56d76111e8ff82fb242a23290da941283/ansible/deploy_aws.yml#L41)):*
<details><summary><i>Previously, mreid-moz (Mark Reid) wrote…</i></summary>
> Nope! I left that in from the first pass where I actually fetched the file. I'll remove it.
</details>
Done.

---

*[ansible/envs/dev.yml, line 28 \[r4\]](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-KLPgvpHZnnycfk4pTsT:-KLRQnv-uSIGogvyYSP-:-499146009) ([raw file](https://github.com/vitillo/telemetry-airflow/blob/38dc9df56d76111e8ff82fb242a23290da941283/ansible/envs/dev.yml#L28)):*
<details><summary><i>Previously, mreid-moz (Mark Reid) wrote…</i></summary>
> No, I'll remove it
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9)*

**Comment by vitillo** [comment]
Reviewed 6 of 6 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[ansible/files/airflow/replace_env.py, line 12 \[r5\]](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-KLRTzyQalINhh53womn:-KLRTzyR-ASdo6ML2gjI:669470083) ([raw file](https://github.com/vitillo/telemetry-airflow/blob/005a7ae0efa2e34beab2a74f7f1d10f8cb28b189/ansible/files/airflow/replace_env.py#L12)):*
> ```Python
> 
> input_file = sys.argv[1]
> if len(sys.argv) > 2:
> ```

Can we add a check for len(sys.argv) > 3?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9)*

**Comment by mreid-moz** [comment]
Review status: 4 of 8 files reviewed at latest revision, 1 unresolved discussion.

---

*[ansible/files/airflow/replace_env.py, line 12 \[r5\]](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-KLRTzyQalINhh53womn:-KLRV9Mxu7lKyQuNUH8O:-499146009) ([raw file](https://github.com/vitillo/telemetry-airflow/blob/005a7ae0efa2e34beab2a74f7f1d10f8cb28b189/ansible/files/airflow/replace_env.py#L12)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> Can we add a check for len(sys.argv) > 3?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9)*

**Comment by vitillo** [comment]
:lgtm:

---

Reviewed 4 of 4 files at r6, 4 of 4 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vitillo/telemetry-airflow/9#-:-KLRX5kfsRb-EZAtbCXa:1417088721)*

**Pull request merged**